Set constraints on the temp table in the Astro Python SDK DAG (#1931)
astronomer/docs: learn/astro-python-sdk.md

To load data from S3 into a SQL Table, you only need to specify the location of the data on S3 and an Airflow connection for the destination SQL table in Snowflake.

Because the content of `orders_data` isn't needed after the DAG is completed, it's specified without a name. When you define a `Table` object without a preexisting name, that table is considered a temporary table. The Astro SDK deletes all temporary tables after you run `aql.cleanup` in your DAG.

> In this example, the temporary table is defined with a schema because Snowflake will merge a temporary table only if that table has constraints. However, this is not a requirement for all databases. To see which databases require temporary tables to have restraints, see the [Astro Python SDK documentation](https://astro-sdk-python.readthedocs.io/en/stable/astro/sql/operators/merge.html#prerequisites).

Author: ```suggestion
In this example, a schema is used to define the temporary table because a temporary table can only be merged in Snowflake when it includes constraints. Not all databases require constraints for temporary tables. For a list of databases that require constraints for temporary tables, see the [Astro Python SDK documentation](https://astro-sdk-python.readthedocs.io/en/stable/astro/sql/operators/merge.html#prerequisites).
```

    output_table=Table(
        conn_id=SNOWFLAKE_CONN_ID,
        output_table=Table(
            conn_id=SNOWFLAKE_CONN_ID,
            columns=[
                sqlalchemy.Column("order_id", sqlalchemy.String(60), primary_key=True),
                sqlalchemy.Column(
                    "customer_id",
                    sqlalchemy.String(60),
                    nullable=False,
                    key="customer_id",
                ),
                sqlalchemy.Column(
                    "purchase_date",
                    sqlalchemy.String(60),
                    nullable=False,
                    key="purchase_date",
                ),
                sqlalchemy.Column(
                    "amount", sqlalchemy.Integer, nullable=False, key="amount"
                ),
            ],
        ),   
)
```

Because the content of `orders_data` isn't needed after the DAG is completed, it's specified without a name. When you define a `Table` object without a preexisting name, that table is considered a temporary table. The Astro SDK deletes all temporary tables after you run `aql.cleanup` in your DAG.

> In this example, the temporary table is defined with a schema because Snowflake will merge a temporary table only if that table has constraints. However, this is not a requirement for all databases. To see which databases require temporary tables to have restraints, see the [Astro Python SDK documentation](https://astro-sdk-python.readthedocs.io/en/stable/astro/sql/operators/merge.html#prerequisites).

Reviewer: ```suggestion
In this example, a schema is used to define the temporary table because a temporary table can only be merged in Snowflake when the temporary table includes constraints. Not all databases require constraints for temporary tables. To determine which databases require constraints for temporary tables, see the [Astro Python SDK documentation](https://astro-sdk-python.readthedocs.io/en/stable/astro/sql/operators/merge.html#prerequisites).
```

Reviewer: Should this information not appear _before_ the example?
